test/stdlib/Random.swift

```diff
@@ -0,0 +1,153 @@
+  where T.Stride: SignedInteger, T.Magnitude: UnsignedInteger {
+    
+  let testRange = 0 ..< 1_000
+  var integerSet: Set<T> = []
+  
+  // min open range
+  let minOpenRange = T.min ..< (T.min + 10)
+  for _ in testRange {
+    let random = T.random(in: minOpenRange)
+    expectTrue(minOpenRange.contains(random))
+    integerSet.insert(random)
+  }
+  expectTrue(integerSet == Set(T.min ..< (T.min + 10)))
+  integerSet.removeAll()
+  
+  // min closed range
+  let minClosedRange = T.min ... (T.min + 10)
+  for _ in testRange {
+    let random = T.random(in: minClosedRange)
+    expectTrue(minClosedRange.contains(random))
+    integerSet.insert(random)
+  }
+  expectTrue(integerSet == Set(T.min ... (T.min + 10)))
+  integerSet.removeAll()
+  
+  // max open range
+  let maxOpenRange = (T.max - 10) ..< T.max
+  for _ in testRange {
+    let random = T.random(in: maxOpenRange)
+    expectTrue(maxOpenRange.contains(random))
+    integerSet.insert(random)
+  }
+  expectTrue(integerSet == Set((T.max - 10) ..< T.max))
+  integerSet.removeAll()
+  
+  // max closed range
+  let maxClosedRange = (T.max - 10) ... T.max
+  for _ in testRange {
+    let random = T.random(in: maxClosedRange)
+    expectTrue(maxClosedRange.contains(random))
+    integerSet.insert(random)
+  }
+  expectTrue(integerSet == Set((T.max - 10) ... T.max))
+}
+
+RandomTests.test("random integers in ranges") {
+  integerRangeTest(Int8.self)
+  integerRangeTest(Int16.self)
+  integerRangeTest(Int32.self)
+  integerRangeTest(Int64.self)
+  integerRangeTest(UInt8.self)
+  integerRangeTest(UInt16.self)
+  integerRangeTest(UInt32.self)
+  integerRangeTest(UInt64.self)
+}
+
+// Random floating points in ranges
+
+func floatingPointRangeTest<T: BinaryFloatingPoint>(_ type: T.Type) 
+  where T.RawSignificand: FixedWidthInteger,
+        T.RawSignificand.Stride: SignedInteger & FixedWidthInteger,
+        T.RawSignificand.Magnitude: UnsignedInteger {
+          
+  let testRange = 0 ..< 1_000
+  
+  // open range
+  let openRange: Range<T> = 0.0 ..< 10.0
+  for _ in testRange {
+    let random = T.random(in: openRange)
+    expectTrue(openRange.contains(random))
+  }
+  
+  // closed range
+  let closedRange: ClosedRange<T> = 0.0 ... 10.0
+  for _ in testRange {
+    let random = T.random(in: closedRange)
+    expectTrue(closedRange.contains(random))
+  }
+}
+
+RandomTests.test("random floating points in ranges") {
+  floatingPointRangeTest(Float.self)
+  floatingPointRangeTest(Double.self)
+  floatingPointRangeTest(Float80.self)
+}
+
+// Random Elements from collection
+
+RandomTests.test("random elements from collection") {
+  let greetings = ["hello", "hi", "hey", "hola", "what's up"]
+  for _ in 0 ..< 1_000 {
+    let randomGreeting = greetings.random()
+    expectNotNil(randomGreeting)
+    expectTrue(greetings.contains(randomGreeting!))
+  }
+}
+
+// uniform distribution
+
+func chi2Test(_ samples: [Double]) -> Bool {
+  let upperBound = 50
+  let numberOfTrials = 500_000
```

Comment:
These two could probably be either a parameter or a global, since they're shared (and need to be the same as) the values in the test below.

In fact, they're both somewhat redundant:  `upperBound == samples.count` and `numberOfTrails = samples.reduce(0, +)`. And, since the DoF is fixed (by the hardcoded confidence interval), it might be worth having it as an assert instead:

```swift
func chi2Test(_ samples: [Double]) -> Bool {
    precondition(samples.count == 50, "confidence interval requires 50 samples")
    let expected = samples.reduce(0, +) / Double(samples.count)
    // ...
}
```